databricks/koalas/namespace.py

```diff
@@ -1863,6 +1864,187 @@ def merge(obj, right: 'DataFrame', how: str = 'inner',
+    """
+    Compute a simple cross tabulation of two factors. By default
+    computes a frequency table of the factors unless an array of values and an
+    aggregation function are passed.
+
+    Parameters
+    ----------
+    index : array-like, Series, or list of arrays/Series
+        Values to group by in the rows.
+    columns : array-like, Series, or list of arrays/Series
+        Values to group by in the columns.
+    rownames : sequence, default None
+        If passed, must match number of row arrays passed.
+    colnames : sequence, default None
+        If passed, must match number of column arrays passed.
+
+    Returns
+    -------
+    DataFrame
+        Cross tabulation of the data.
+
+    Examples
+    --------
+    >>> from databricks.koalas.config import set_option, reset_option
+    >>> set_option("compute.ops_on_diff_frames", True)
+    >>> a = np.array(["foo", "foo", "foo", "foo", "bar", "bar",
+    ...               "bar", "bar", "foo", "foo", "foo"], dtype=object)
+    >>> b = np.array(["one", "one", "one", "two", "one", "one",
+    ...               "one", "two", "two", "two", "one"], dtype=object)
+
+    >>> ks.crosstab(a, b).sort_index()  # doctest: +NORMALIZE_WHITESPACE
+    0    one  two
+    0
+    bar    3    1
+    foo    4    3
+
+    >>> ks.crosstab(a, b, rownames=['koalas'], colnames=['hello']).sort_index()
+    ... # doctest: +NORMALIZE_WHITESPACE
+    hello   one  two
+    koalas
+    bar       3    1
+    foo       4    3
+
+    >>> reset_option("compute.ops_on_diff_frames")
+    """
+    if not isinstance(index, (np.ndarray, Series, list)):
+        raise ValueError("index should be one of `np.ndarray`, `Series`")
+    if not isinstance(columns, (np.ndarray, Series, list)):
+        raise ValueError("columns should be one of `np.ndarray`, `Series`")
+
+    # convert types of parameter `index` and `columns` to list.
+    if not isinstance(index, list):
+        if isinstance(index, np.ndarray):
+            index = [Series(index)]
+        else:
+            index = [index]
+    if not isinstance(columns, list):
+        if isinstance(columns, np.ndarray):
+            columns = [Series(columns)]
+        else:
+            columns = [columns]
+
+    # since there is a high possibility duplicated index and column names,
+    # we just make tmporal names for all index & columns
+    # seems like pandas is also doing similar internally
+    #
+    # >>> pd.crosstab([a, a, a, a, a], [c, c, c, c])
+    # col_0                         dull shiny
+    # col_1                         dull shiny
+    # col_2                         dull shiny
+    # col_3                         dull shiny
+    # row_0 row_1 row_2 row_3 row_4
+    # bar   bar   bar   bar   bar      2     2
+    # foo   foo   foo   foo   foo      3     4
+    tmp_index_names = ['row_{}'.format(i) for i in range(len(index))]
+    tmp_columns_names = ['col_{}'.format(i) for i in range(len(columns))]
+
```

Comment:
I faced very wired problem at this point.

`tmp_index_names` and `tmp_columns_names` returns `['row_id']` and `['col_id']`

![스크린샷 2019-11-29 오후 4 30 24](https://user-images.githubusercontent.com/44108233/69852796-8c9bc780-12c8-11ea-8e12-a757ca6e77ab.png)
![KakaoTalk_Photo_2019-11-29-16-52-03](https://user-images.githubusercontent.com/44108233/69852827-9f160100-12c8-11ea-9a98-7b51525c33c7.png)



but i expected `['row_0', 'row_1' ...]` and `['col_0', 'col_0' ...]`.

i can't get it why they return like this.

and even they works well in python interpreter

![스크린샷 2019-11-29 오후 5 00 27](https://user-images.githubusercontent.com/44108233/69853266-c15c4e80-12c9-11ea-925b-65e25308ad74.png)


anyone have an idea of this wired behavior?? (or maybe this is only my local problem ?)

Reply:
@ueshin @HyukjinKwon could you take a look at this issue once if you available?? 😢

Reply:
![스크린샷 2019-11-29 오후 5 44 01](https://user-images.githubusercontent.com/44108233/69855957-ee136480-12cf-11ea-9a7b-80ee20c99da7.png)

![스크린샷 2019-11-29 오후 5 44 33](https://user-images.githubusercontent.com/44108233/69855976-f075be80-12cf-11ea-94bb-ceba39cbcd2f.png)

maybe it is not an koalas issue, but it's really wired 😅

Reply:
oops. it was reproduced here in travisCI also.

<img width="986" alt="스크린샷 2019-11-29 오후 5 48 20" src="https://user-images.githubusercontent.com/44108233/69856231-81e53080-12d0-11ea-972f-635237c44897.png">
https://travis-ci.com/databricks/koalas/jobs/261530945#L2443

Reply:
It's because `range` is defined in this file `namespace.py`. You could:

```python

# Keep built-in function 'range'
_range = range


def range(start: int,
          end: Optional[int] = None,
          step: int = 1,
          num_partitions: Optional[int] = None) -> DataFrame:
...

    tmp_index_names = ['row_{}'.format(i) for i in _range(len(index))]
    tmp_columns_names = ['col_{}'.format(i) for i in _range(len(columns))]

```

Reply:
oops. why i did't catch that?!

thanks for the catch!! 👍